environment.cpp

```diff
@@ -116,9 +122,20 @@ std::vector<double> create_n_inputs(std::uniform_real_distribution<double> dist,
 
 double dummy_function(std::vector<double> input)
 {
-  return input[0]+1;
+  return input[0]+123;
 }
 
+void switch_env_function(environment &e)
+{
+  if(are_same_env_functions(e.get_current_function(), e.get_env_function_A())){
```

Comment:
I am trying to see if we can do better than this, since in terms of performance this might be costly.